src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -318,102 +336,134 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             }
         }
 
-        private int FindEntry(TKey key)
+        public unsafe bool TryGetValue(TKey key, [MaybeNullWhen(false)] out TValue value)
```

Comment:
What is the performance going to be for larger TValue? E.g. `ValueTuple<object, object, object, object>` ? I would expect that e.g. `ContainsKey` will regress for this case.

I liked your previous approach where this returned `ref TValue` better.

Reply:
Changed